drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/referral_postings_loader.rb

```diff
@@ -0,0 +1,229 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::AcHmis::Importers::Loaders
+  class ReferralPostingsLoader < BaseLoader
+    def perform
+      init_enrollment_ids_by_referral_id
+
+      # destroy existing records and re-import
+      referral_scope.destroy_all if clobber
+
+      import_referral_records
+      import_referral_posting_records
+      import_referral_household_members_records
+      assign_unit_occupancies
+    end
+
+    POSTINGS_FILENAME = 'ReferralPostings.csv'.freeze
+    HOUSEHOLD_MEMBERS_FILENAME = 'ReferralHouseholdMembers.csv'.freeze
+
+    def runnable?
+      super &&
+      reader.file_present?(POSTINGS_FILENAME) &&
+      reader.file_present?(HOUSEHOLD_MEMBERS_FILENAME)
+    end
+
+    def table_names
+      [
+        HmisExternalApis::AcHmis::ReferralHouseholdMember.table_name,
+        HmisExternalApis::AcHmis::ReferralPosting.table_name,
+      ]
+    end
+
+    protected
+
+    def import_referral_records
+      ar_import(
+        referral_class,
+        build_referral_records,
+        on_duplicate_key_update: { conflict_target: :identifier, columns: :all },
+      )
+    end
+
+    def import_referral_posting_records
+      ar_import(
+        HmisExternalApis::AcHmis::ReferralPosting,
+        build_posting_records,
+        on_duplicate_key_update: { conflict_target: :identifier, columns: :all },
+      )
+    end
+
+    def import_referral_household_members_records
+      ar_import(
+        HmisExternalApis::AcHmis::ReferralHouseholdMember,
+        build_household_member_records,
+        on_duplicate_key_update: { conflict_target: [:client_id, :referral_id], columns: :all },
+      )
+    end
+
+    def supports_upsert?
+      true
+    end
+
+    def posting_rows
+      reader.rows(POSTINGS_FILENAME)
+    end
+
+    def household_member_rows
+      reader.rows(HOUSEHOLD_MEMBERS_FILENAME)
+    end
+
+    def referral_class
+      HmisExternalApis::AcHmis::Referral
+    end
+
+    def build_referral_records
+      posting_rows.map do |row|
+        referral_id = row_value(row, field: 'REFERRAL_ID')
+        referral_class.new(
+          identifier: referral_id,
+          enrollment_id: referral_enrollment_id(referral_id),
```

Comment:
```suggestion
```

Sorry, this is confusingly named, that's my bad. The `enrollment_id` col stores a reference to the enrollment that the referral _came from_, which only gets set for referrals that are made within the HMIS. For LINK-originating referrals, it's always null.

Instead, the "link" to the enrollment is stored by way of `HouseholdID` column on the `hmis_external_referral_postings` table.
So:
1. For accepted referrals, we ensure that we can find the enrollment by ENROLLMENT_ID, and then store the enrollment's household ID on the referral posting record.
2. For accepted pending referrals, we create an enrollment for each household member, and then store the household ID on the referral posting record.

The reason for using household ids, rather than enrollment ids, is because the household membership my change but it still needs to remain "linked" to the posting.

Reply:
Thanks for clarifying this! This all sounds familiar, I think I misremembered how it was used 🤦 
I added comments to the model classes to document it and removed the enrollment reference from the referrals.